src/views/CommerceView.vue

```diff
@@ -61,18 +63,13 @@ export default {
     }
   },
   watch: {
-    '$route.query.filter': {
-      handler: function (filter) {
-        this.createFilter()
-        this.setProducts(this.$route.params.slug, filter)
-      },
-      deep: true,
-      immediate: true,
-    },
```

Comment:
Why remove the watcher?

Reply:
because it was causing flicker, doing 3 requests instead of 1

Reply:
Ah okay, but now if I select a sort by on the sidebar, does the UI change acc. to that?

Reply:
yes, everything should be ok -> https://jakub-jarek-edgio-vue3-example-default.layer0-limelight.link

Reply:
yup, and fast!